src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSLambdaHttpUtils.cs

```diff
@@ -0,0 +1,134 @@
+
+            switch (input)
+            {
+                case APIGatewayProxyRequest request:
+                    httpScheme = AWSLambdaUtils.GetHeaderValues(request, HeaderXForwardedProto)?.FirstOrDefault();
+                    httpTarget = string.Concat(request.Path ?? string.Empty, ConstructQueryString(request));
+                    httpMethod = request.HttpMethod;
+                    var hostHeader = AWSLambdaUtils.GetHeaderValues(request, HeaderHost)?.FirstOrDefault();
+                    (hostName, hostPort) = GetHostAndPort(httpScheme, hostHeader);
+                    break;
+                case APIGatewayHttpApiV2ProxyRequest requestV2:
+                    httpScheme = AWSLambdaUtils.GetHeaderValues(requestV2, HeaderXForwardedProto)?.FirstOrDefault();
+                    httpTarget = string.Concat(requestV2?.RawPath ?? string.Empty, ConstructQueryString(requestV2));
+                    httpMethod = requestV2?.RequestContext?.Http?.Method;
+                    var hostHeaderV2 = AWSLambdaUtils.GetHeaderValues(requestV2, HeaderHost)?.FirstOrDefault();
+                    (hostName, hostPort) = GetHostAndPort(httpScheme, hostHeaderV2);
+                    break;
+            }
+
+            tags.AddTagIfNotNull(SemanticConventions.AttributeHttpScheme, httpScheme);
+            tags.AddTagIfNotNull(SemanticConventions.AttributeHttpTarget, httpTarget);
+            tags.AddTagIfNotNull(SemanticConventions.AttributeHttpMethod, httpMethod);
+            tags.AddTagIfNotNull(SemanticConventions.AttributeNetHostName, hostName);
+            tags.AddTagIfNotNull(SemanticConventions.AttributeNetHostPort, hostPort);
+
+            return tags;
+        }
+
+        internal static void SetHttpTagsFromResult(Activity activity, object result)
+        {
+            switch (result)
+            {
+                case APIGatewayProxyResponse response:
+                    activity.SetTag(SemanticConventions.AttributeHttpStatusCode, response.StatusCode);
+                    break;
+                case APIGatewayHttpApiV2ProxyResponse responseV2:
+                    activity.SetTag(SemanticConventions.AttributeHttpStatusCode, responseV2.StatusCode);
+                    break;
+            }
+        }
+
+        internal static string ConstructQueryString(APIGatewayProxyRequest request)
+        {
+            if (request.MultiValueQueryStringParameters == null)
+            {
+                return string.Empty;
+            }
+
+            var items = new List<string>();
+            foreach (var name in request.MultiValueQueryStringParameters.Keys)
+            {
+                // Multiple values for the same parameter will be added to query
+                // as ampersand separated: name=value1&name=value2
+                foreach (var value in request.MultiValueQueryStringParameters[name])
+                {
+                    items.Add(string.Concat(name, "=", HttpUtility.UrlEncode(value)));
```

Comment:
Lets keep it simple and do not encode both. Removed encoding of the value.

Reply:
While simplicity is good, correctness is more important. We have to check which of these are already URL-decoded and re-encode them. Otherwise this will break if there are special characters like #/?/=/& in the name or value.

Reply:
AWS checks show that values not encoded and we need to encode `name` and `value`. So, now both `name` and `value `are encoded.